Sources/CacheAdvance/CacheReader.swift

```diff
@@ -75,16 +69,7 @@ final class CacheReader {
             return message
 
         case .emptyRead:
-            guard offsetInFileAtEndOfNewestMessage < startingOffset else {
-                // We started reading before the offset of the end of the newest message, therefore we expected a message to be read. We instead read an empty space, meaning that the file is corrupt.
-                throw CacheAdvanceError.fileCorrupted
-            }
-
-            // We know the next message is at the end of the file header. Let's seek to it.
-            try reader.seek(to: FileHeader.expectedEndOfHeaderInFile)
-
-            // We know there's a message to read now that we're at the start of the file.
-            return try nextEncodedMessage()
+            return nil
```

Comment:
can you add a code comment here to help me understand what case we're hitting here / why it is expected that we return `nil` here?

Reply:
here we return nil to stop reading